dd-sdk-android/src/main/kotlin/com/datadog/android/DatadogConfig.kt

```diff
@@ -442,6 +444,27 @@ private constructor(
             return this
         }
 
+        private fun sanitizeHosts(hosts: List<String>): List<String> {
+            return hosts.mapNotNull {
+                if (it.matches(URL_REGEX)) {
+                    try {
+                        val parsedUrl = URL(it)
+                        devLogger.w(
+                            "You are using an url: $it for declaring the first " +
+                                "party hosts. You should use instead a valid" +
+                                " host name: ${parsedUrl.host}."
+                        )
+                        parsedUrl.host
+                    } catch (e: MalformedURLException) {
+                        devLogger.e("The url: $it is malformed. It will be dropped", e)
+                        null
+                    }
+                } else {
```

Comment:
While we're here we can check if the host matches a `HOST_REGEX` (something like `^(([a-zA-Z0-9]|[a-zA-Z0-9][a-zA-Z0-9\-]*[a-zA-Z0-9])\.)*([A-Za-z0-9]|[A-Za-z0-9][A-Za-z0-9\-]*[A-Za-z0-9])$`) or an `IP_REGEX` (something like `(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)(\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)){3}`)
That way if the host doesn't match any known pattern we can print an error and ignore it.